src/main/java/net/minestom/server/utils/time/TimeUnit.java

```diff
@@ -2,6 +2,10 @@
 
 import net.minestom.server.MinecraftServer;
 
+/**
+ * @deprecated Replaced by {@link java.time.temporal.TemporalUnit}
+ */
+@Deprecated(forRemoval = true)
 public enum TimeUnit {
```

Comment:
I believe that we should keep this as a class to expose all our time units. Otherwise we will end up using `ChronoUnit` and `Tick` separately.

Reply:
This enum could be made to implement `TemporalUnit`, delegating the methods to both `Tick` and `ChronoUnit` respectively. However, I'm not a massive fan of this option as a long-term solution.

Reply:
I think duplicating code (in this case keeping the enum instead of `ChronoUnit`) isn't a good thing, personally i think nothing wrong with using the two beside each other.

Reply:
The issue is that most people will not know that the two exist. I think that it makes sense to list everything somewhere as it would be officialy supported

Reply:
Yeah i see... Then can we settle with a compromise so that the `TimeUnit` isn't going to be for removal (but still deprecated) and in the deprecation notice include `Tick` and `ChronoUnit`?

Reply:
Maybe add a wiki page too, long before removal, that way it really shouldn't cause issues.

Reply:
I don't really see why adding constants in TimeUnit is an issue, it is in the utils package and could be used to list all the officially supported units

Reply:
Oh wait... I thought about a different implementation of this... nevermind; so you mean like this?
```java
public enum TimeUnit {
    DAY(ChronoUnit.DAYS);
    //...

    private final TemporalUnit temporalUnit;

    TimeUnit(TemporalUnit temporalUnit) {
        this.temporalUnit = temporalUnit;
    }

    public TemporalUnit getTemporalUnit() {
        return temporalUnit;
    }
}
```

Reply:
I said class constants, so
```java
public static final TemporalUnit DAY = ChronoUnit.DAYS;
public static final TemporalUnit TICK = Tick.TICKS;
```

Reply:
Ah okay, i don't know why i wanted to stick with the enum... I'm fine with constants

Reply:
Oh i know why i wanted to stick with the enum, because of deprecation so it won't break existing projects, but if we going to use class instead of enum there is no point of keeping deprecated methods i marked for removal previously.